src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -98,20 +63,23 @@ export default class SimidMapCreative extends BaseSimidCreative {
         return;
     }
     this.buttonLabel_ = adParams[AdParamKeys.BUTTON_LABEL]; 
-    this.searchQuery_ = adParams[AdParamKeys.SEARCH_QUERY];
-    this.markerImage_ = adParams[AdParamKeys.MARKER];
-    this.userCoordinates_ = adParams[AdParamKeys.COORDINATES];
+    const query = adParams[AdParamKeys.SEARCH_QUERY];
```

Comment:
The function is now doing something other than validating and parsing ad params.  It is setting up the simidMap object.

Ideally you can just have it validate and parse here.  Then when you do set up the simid map object you can trust that the parameters are correct.

Reply:
Updated this based off of Kaleab's comment below